Nim compiler is unable to build under gcc 4.1 on linux

Hi there.

I'm using nim devel, currently at commit e2e663a1438

According to this page:

https://github.com/nim-lang/Nim#compiling

Nim should be able to compile on gcc 3.x or later.

However, during building of Nim compiler with gcc 4.1, I get this error, while attempting to compile:

stdlib_sha1.nim.c:(.text+0x2765): undefined reference to `__builtin_bswap32'

Based on what I can tell, and also based on this stackoverflow article:

https://stackoverflow.com/questions/14713807/compilation-error-related-to-builtin-bswap32

"__builtin_bswap32" support was only added to gcc in v4.3

So I think this means that either:

1. The documentation on Nim's github page needs to refer to "gcc v4.3 or later", or
2. The compile process for Nim needs to use an alternative to "__builtin_bswap32"

Can someone else please try checking if Nim still compiles on "gcc 3.x/later", or whether that's no longer the case?

...

My use case is attempting to install Nim from source, under a "Debian 4 i386" environment. Which is severely out of date, but which is out of my control, but where I still need to target for the time being in my work dev. I'm currently using Python 2.4 in the environment for services (which comes with that distro release, as well as a lot of libraries for Python 2.4), but I'm also looking into alternatives.

--- a/readme.md
+++ b/readme.md
@@ -51,8 +51,8 @@ Nim programming language. Those C sources are available within the
 
 Next, to build from source you will need:
 
-  * A C compiler such as ``gcc`` 3.x/later or an alternative such as ``clang``,
-    ``Visual C++`` or ``Intel C++``. It is recommended to use ``gcc`` 3.x or
+  * A C compiler such as ``gcc`` 5.x/later or an alternative such as ``clang``,
+    ``Visual C++`` or ``Intel C++``. It is recommended to use ``gcc`` 5.x or
     later.
   * Either ``git`` or ``wget`` to download the needed source repositories.
   * The ``build-essential`` package when using ``gcc`` on Ubuntu (and likely

